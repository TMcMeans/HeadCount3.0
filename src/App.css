@import url('https://fonts.googleapis.com/css?family=Poppins');

.title,
.heading {
  font-family: 'Poppins', sans-serif;
}

.title {
  font-size: 2.7em;
}

.heading {
  font-size: 1.1em;
}

hr {
  height: 50px;
  width: 100%;
  margin-top: 0;
  background: #4a314d;
  border-bottom: 6px solid #f0af4d;
  border-top: none;
  border-left: none;
  border-right: none;
}

.header {
  width: 85%;
  margin: 0 auto;
  margin-bottom: 50px;
}
