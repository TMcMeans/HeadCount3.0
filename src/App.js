import React, { Component } from 'react';
import kinderData from './data/kindergartners_in_full_day_program.js';

import './App.css';
import DistrictRepository from './helper.js';
import { CardContainer } from './components/CardContainer';

class App extends Component {
  constructor() {
    super();
    this.state = {
      repository: new DistrictRepository(kinderData)
    };
  }

  componentDidMount = () => {};

  render() {
    const { findAllMatches } = this.state.repository;
    return (
      <div>
        <h1>Welcome to HeadCount 3.0</h1>
        <CardContainer findAllMatches={findAllMatches} />
      </div>
    );
  }
}

export default App;
