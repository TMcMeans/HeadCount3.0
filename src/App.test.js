import React from 'react';
import { shallow, mount } from 'enzyme';
import App from './App';

import kinderData from './data/kindergartners_in_full_day_program.js';
import DistrictRepository from './helper.js';

describe('App', () => {
  it('should match snapshot with all html elements rendered correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state with district repository data and properties when App is mounted', () => {
    const wrapper = shallow(<App />);
    const repository = new DistrictRepository(kinderData);
    const expectedState = {
      repository: new DistrictRepository(kinderData),
      inputSearchName: '',
      compareSchoolSelections: []
    };

    // expect(wrapper.state()).toEqual(expectedState);
    expect(wrapper.state().equals(expectedState)).toBe(true);
  });

  it('should update state with inputSearchName when filterCards is called', () => {
    const mockInputName = 'colorado';
    const wrapper = mount(<App />);

    wrapper.instance().filterCards(mockInputName);
    expect(wrapper.state('inputSearchName')).toEqual(mockInputName);
  });

  it('should update state to add school names to compareSchoolSelections array when handleCompareSelections is called', () => {
    const wrapper = mount(<App />);
    const school1 = 'COLORADO';
    const school2 = 'ADAMS COUNTY 14';

    const mockCompareSelections = ['ADAMS COUNTY 14', 'COLORADO'];

    wrapper.instance().handleCompareSelections(school1);
    wrapper.instance().handleCompareSelections(school2);

    expect(wrapper.state('compareSchoolSelections')).toEqual([
      'ADAMS COUNTY 14',
      'COLORADO'
    ]);
  });

  it('should remove an added schoolName from state once removeCompraredSelections is called', () => {
    const wrapper = mount(<App />);
    const school1 = 'COLORADO';
    const school2 = 'ADAMS COUNTY 14';

    const mockCompareSelections = ['ADAMS COUNTY 14', 'COLORADO'];

    wrapper.instance().handleCompareSelections(school1);
    wrapper.instance().handleCompareSelections(school2);

    wrapper.instance().removeCompareSelections('ADAMS COUNTY 14');

    expect(wrapper.state('compareSchoolSelections')).toEqual(['COLORADO']);
  });
});
