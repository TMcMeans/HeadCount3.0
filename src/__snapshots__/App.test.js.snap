// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot with all html elements rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Welcome to HeadCount 3.0
</h1>,
        <SearchInput
          filterCards={[Function]}
/>,
        <Unknown
          findAllMatches={[Function]}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome to HeadCount 3.0",
        },
        "ref": null,
        "rendered": "Welcome to HeadCount 3.0",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "filterCards": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "findAllMatches": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Welcome to HeadCount 3.0
</h1>,
          <SearchInput
            filterCards={[Function]}
/>,
          <Unknown
            findAllMatches={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to HeadCount 3.0",
          },
          "ref": null,
          "rendered": "Welcome to HeadCount 3.0",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "filterCards": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "findAllMatches": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
